src/parser.cc

```diff
@@ -1001,30 +1001,47 @@ bool readXml(tinyxml2::XMLElement *_xml, ElementPtr _sdf, Errors &_errors)
         }
 
         sdf::ElementPtr topLevelElem;
-        bool isModel{false};
-        bool isActor{false};
-        if (includeSDF->Root()->HasElement("model"))
+        int countTopLevelElements = 0;
+        for (const auto & elementType : {"model", "actor", "light"})
         {
-          topLevelElem = includeSDF->Root()->GetElement("model");
-          isModel = true;
+          if (includeSDF->Root()->HasElement(elementType))
+          {
+            countTopLevelElements++;
+            topLevelElem = includeSDF->Root()->GetElement(elementType);
+          }
         }
-        else if (includeSDF->Root()->HasElement("actor"))
+
+        if (nullptr == topLevelElem)
         {
-          topLevelElem = includeSDF->Root()->GetElement("actor");
-          isActor = true;
+          _errors.push_back({ErrorCode::ELEMENT_INVALID,
+              "Failed to find top level <model> / <actor> / <light> for "
+              "<include>\n"});
+          continue;
         }
-        else if (includeSDF->Root()->HasElement("light"))
+
+        // Check if more than one type of top-level element
+        if (countTopLevelElements > 1)
         {
-          topLevelElem = includeSDF->Root()->GetElement("light");
+          _errors.push_back({ErrorCode::ELEMENT_INVALID,
```

Comment:
Per our discussion the other day, I think we should make this a warning for libsdformat11. Since we don't have a way to include warnings in `sdf::Errors`, we would just print a warning message via `sdfwarn`.

Reply:
After some rebasing, I now have this added.